provider/core/src/resource.rs

```diff
@@ -92,19 +126,42 @@ impl ResourceKey {
     }
 
     const fn check_path_syntax(path: &str) -> Result<(), ()> {
+        let path = path.as_bytes();
+
+        // Start and end of the untagged part
+        let start = leading_tag!().len();
+        let end = path.len() - trailing_tag!().len();
+
+        // Check tags
+        if start > end {
+            return Err(());
+        }
```

Comment:
Nit: Move this to the top of the function and make it

```suggestion
        if path.len() <= leading_tag!().len() + trailing_tag!().len() {
            return Err(());
        }
```

Reply:
Why? That seems way harder to read

Reply:
Ok but only because this panics if the string is shorter than `trailing_tag!()`.

Reply:
Correct, and in general because we know that `leading_tag!().len() + trailing_tag!().len()` resolves to a constant value, whereas `end` is computed at runtime and could overflow, etc.